### ISSUE TITLE
Can't remove/change fade from white transition when new image is loaded

### ISSUE THREAD

**Issue by Shuky** [Norserium/vue-advanced-cropper#135]
Fantastic library, but the cropper has a fade in animation when loading a new image that cannot be removed or overridden.

**Comment by Norserium** [comment]
@Shuky, hello! Are you aware, that the image changing will reset the coordinates? I assume that it's the next issue, that you will encounter.

**Comment by Shuky** [comment]
@Norserium hi!
Actually already took care of that by saving to a tmp and updating the cropper on the ready event.

For now I actually hacked a version
Where I manipulate the sourceCanvas field and made a modified version of the “onChangeImage” function that doesn’t set imageLoaded to false.

seems like it did the trick

**Comment by Norserium** [comment]
Just for curiosity, do the setting of `transitionTime` to `0` fix your issue? Don't even event use it in production, because I will remove this property soon, but it will be helpful if you try it now.

**Comment by trpx** [comment]
Hi folks, I am encountering the same issue (my use-case is the same - I am changing saturation etc - i.e. implementing a more feature-complete image editor based on this great component).
And setting transitionTime to 0 doesn't fix this completely, there is still a noticeable transition to white and back (although faster).

The white background issue goes away if you set transition: none styles for background and foreground (as far as I can tell after a quick investigation, there is no idiomatic way to do this for now, so external css is one option), but the image keeps disappearing during reloading even if you remove the whitening transition with css and transitionTime=0.
It would be so great to be able to change the underlying image from the outside of the component without such side-effects.

What would be the best way to apply such alterations to the image without resetting the coordinates/visible area etc?
For now I am considering hacking into source to implement a hook for arbitrary manipulations. Is there any function which manipulates the underlying image pixels which can be used for inspiration? Or is there a better way?

And **thank you for this awesome library**, it is the best of the kind for sure, great work.

**Comment by Norserium** [comment]
@trpx, hello!

> Is there any pixel-manipulating function (within the component) which can be used for inspiration? 

What do you mean by "pixel-manipulating function"?

**Comment by Norserium** [comment]
@trpx, @Shuky, I agree that the swapping image procedure should be changed, because it's obviously too rigid now. You can't affect transitions. You can't preserve coordinates, visible area or image transforms.  I do agree, but I'm not sure that I have spare time to redesign this complicated aspect of the library right now.

It's the bad news. But I have good news at last. There is the nice **workaround** that can be used to smoothly change image without any side effects. [Take a look](https://codesandbox.io/s/vue-advanced-cropper-workaround-u7tv6?file=/src/App.vue).

**Comment by bgomes90** [comment]
same problem here... any solution?

**Comment by Norserium** [comment]
@bgomes90, there is [the workaround](https://github.com/Norserium/vue-advanced-cropper/issues/135#issuecomment-830052658) only now. 

I suppose that the regular solution will be available in the next major release on the analogy of [react-advanced-cropper method](https://norserium.github.io/react-advanced-cropper/docs/components/Cropper#setimageimage):
```ts
cropper.setImage(image);
```

**Comment by Norserium** [comment]
@bgomes90, if you need to preserve the transparency just set the `image/png` during the transformation to `DataURL` or `Blob`. [The example](https://codesandbox.io/s/vue-advanced-cropper-change-background-1y71fk?file=/src/App.vue).

**Comment by Norserium** [comment]
> I'm currently adding effects to the existing image (contrast, saturation, etc.) and would love to be able to make the transition smoother when I switch the cropper src image.

@Shuky, @trpx,  just for your information. I assume that in the next major release it will be implemented in [this manner](https://norserium.github.io/react-advanced-cropper/docs/tutorials/image-editor). Any feedback is appreciated.
